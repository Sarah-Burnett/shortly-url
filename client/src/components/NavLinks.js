import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";
import { Button, PrimaryButton, white, grey, violet, desktop } from "./Styles";
import { useUser } from "../hooks/useUser";

const Links = styled.div`
	display: flex;
	@media (max-width: ${desktop}) {
		position: absolute;
		z-index: 101;
		top: 10vh;
		left: 1vw;
		height: 89vh;
		width: 98vw;
		flex-direction: column;
		background: ${violet};
		padding: 2vh 2vw;
		border-radius: 15px;
		.resourcesBtn:after {
			content: "";
			display: block;
			width: 70vw;
			margin: auto;
			height: 10vh;
			border-bottom: 2px solid ${grey};
		}
		button {
			display: block;
			color: ${white};
			a {
				color: ${white};
			}
			margin: 5px 0;
		}
		div {
			display: flex;
			flex-direction: column;
			justify-content: space-evenly;
			align-items: center;
		}
		div:first-child {
			height: 60%;
		}
		div:nth-of-type(2) {
			button {
				width: 70vw;
				margin: 2vh;
				border-radius: 30px;
			}
		}
	}
	@media (min-width: ${desktop}) {
		position: relative;
		flex-basis: 75vw;
		flex-direction: row;
		justify-content: space-between;
		align-items: center;
		height: 10vh;
		width: auto;
		button {
			margin: 2vh;
		}
	}
`;

const NavLinks = ({ toggleMobNav = () => {} }) => {
	const {
		user: { isAuth },
	} = useUser();
	return (
		<Links onClick={toggleMobNav}>
			<div>
				<Button>Features</Button>
				<Button>Pricing</Button>
				<Button className="resourcesBtn">Resources</Button>
			</div>
			<div data-testid="authBtns">
				{!isAuth && (
					<Link to="/login">
						<Button>Login</Button>
					</Link>
				)}
				{!isAuth && (
					<Link to="/signup">
						<PrimaryButton>Sign Up</PrimaryButton>
					</Link>
				)}
				{isAuth && (
					<Link to="/logout">
						<PrimaryButton>Logout</PrimaryButton>
					</Link>
				)}
			</div>
		</Links>
	);
};

export default NavLinks;
