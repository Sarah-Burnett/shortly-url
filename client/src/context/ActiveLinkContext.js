import React, { useState, createContext } from 'react';

export const ActiveLinkContext = createContext();

export const ActiveLinkProvider = (props) => {
    const [ activeLink, setActiveLink] = useState({
        activeLink: [],
        process: false
    });

    return (
        <ActiveLinkContext.Provider value={[activeLink, setActiveLink]}>
            {props.children}
        </ActiveLinkContext.Provider>
    )
};