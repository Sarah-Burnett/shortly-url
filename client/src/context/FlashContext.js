import React, { useState, createContext } from 'react';

export const FlashContext = createContext();

export const FlashProvider = (props) => {
    const [ flash, setFlash ] = useState({
        flashMsg: "",
        isFlashVisible: false,
        isAutoHideFlash: true,
    });
    return (
        <FlashContext.Provider value={[flash, setFlash]}>
            {props.children}
        </FlashContext.Provider>
    )
};