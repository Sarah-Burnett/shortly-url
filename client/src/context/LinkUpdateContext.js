import React, { useState, createContext } from 'react';

export const LinkUpdateContext = createContext();

export const LinkUpdateProvider = (props) => {
    const [ linkUpdate, setLinkUpdate] = useState();
    return (
        <LinkUpdateContext.Provider value={[linkUpdate, setLinkUpdate]}>
            {props.children}
        </LinkUpdateContext.Provider>
    )
};