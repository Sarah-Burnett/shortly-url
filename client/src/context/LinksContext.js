import React, { useState, createContext } from 'react';

export const LinksContext = createContext();

export const LinksProvider = (props) => {
    const [ links, setLinks] = useState([]);

    return (
        <LinksContext.Provider value={[links, setLinks]}>
            {props.children}
        </LinksContext.Provider>
    )
};