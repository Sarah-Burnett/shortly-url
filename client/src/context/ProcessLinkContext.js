import React, { useState, createContext } from 'react';

export const ProcessLinkContext = createContext();

export const ProcessLinkProvider = (props) => {
    const [ processLink, setProcessLink] = useState({
        link: [],
        process: false
    });
    return (
        <ProcessLinkContext.Provider value={[processLink, setProcessLink]}>
            {props.children}
        </ProcessLinkContext.Provider>
    )
};