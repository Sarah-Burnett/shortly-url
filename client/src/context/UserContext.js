import React, { useState, createContext } from 'react';

export const UserContext = createContext();

export const UserProvider = (props) => {
    const [ user, setUser] = useState({
        isAuth: false,
        firstName: undefined,
        id: undefined,
        token: undefined,
    });
    
    return (
        <UserContext.Provider value={[user, setUser]}>
            {props.children}
        </UserContext.Provider>
    )
};