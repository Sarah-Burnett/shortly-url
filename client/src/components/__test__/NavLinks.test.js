import React, { useState } from 'react';
import ReactDOM from 'react-dom';
import NavLinks from '../NavLinks';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { UserContext } from '../../context/UserContext';
import { BrowserRouter as Router } from 'react-router-dom';
afterEach(cleanup);

const NavLinksWrapper = ({isAuth = false}) => {
    const [ user, setUser] = useState({isAuth});
    return (
        <UserContext.Provider value={[user, setUser]}>
            <Router>
                <NavLinks/>
            </Router>
        </UserContext.Provider>
    )
}

it('renders NavLinks without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NavLinksWrapper/>, div);
})

it('renders not-authenticated nav links correctly', () => {
    const { getByTestId } = render(<NavLinksWrapper/>);
    expect(getByTestId('authBtns').childNodes.length).toBe(2);
});

it('nav links not-authenticated matches snapshot', () => {
    const tree = renderer.create(<NavLinksWrapper/>).toJSON();
    expect(tree).toMatchSnapshot();
})

it('renders authenticated nav links correctly', () => {
    const { getByTestId } = render(<NavLinksWrapper isAuth/>);
    expect(getByTestId('authBtns').childNodes.length).toBe(1);
});

it('nav links authenticated matches snapshot', () => {
    const tree = renderer.create(<NavLinksWrapper isAuth/>).toJSON();
    expect(tree).toMatchSnapshot();
})