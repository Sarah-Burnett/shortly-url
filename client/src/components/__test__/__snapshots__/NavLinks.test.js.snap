// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav links authenticated matches snapshot 1`] = `
<div
  className="sc-fzplWN fTyIJW"
  onClick={[Function]}
>
  <div>
    <button
      className="sc-AxjAm iwCWZU"
    >
      Features
    </button>
    <button
      className="sc-AxjAm iwCWZU"
    >
      Pricing
    </button>
    <button
      className="sc-AxjAm iwCWZU resourcesBtn"
    >
      Resources
    </button>
  </div>
  <div
    data-testid="authBtns"
  >
    <a
      href="/logout"
      onClick={[Function]}
    >
      <button
        className="sc-AxirZ fTDgvF"
      >
        Logout
      </button>
    </a>
  </div>
</div>
`;

exports[`nav links not-authenticated matches snapshot 1`] = `
<div
  className="sc-fzplWN fTyIJW"
  onClick={[Function]}
>
  <div>
    <button
      className="sc-AxjAm iwCWZU"
    >
      Features
    </button>
    <button
      className="sc-AxjAm iwCWZU"
    >
      Pricing
    </button>
    <button
      className="sc-AxjAm iwCWZU resourcesBtn"
    >
      Resources
    </button>
  </div>
  <div
    data-testid="authBtns"
  >
    <a
      href="/login"
      onClick={[Function]}
    >
      <button
        className="sc-AxjAm iwCWZU"
      >
        Login
      </button>
    </a>
    <a
      href="/signup"
      onClick={[Function]}
    >
      <button
        className="sc-AxirZ fTDgvF"
      >
        Sign Up
      </button>
    </a>
  </div>
</div>
`;
